### PR TITLE
docs: Add workload storage documentation

### PR DESCRIPTION
**Pull request by lucieleonard**
Fixes: #400

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (NicolasT) [inline .vscode/settings.json]: Don't think we want these files to be checked in. Add `.vscode/` to `.gitignore` instead. We could consider adding a `.editorconfig`, but even then ;)

File: docs/usage/storage.rst [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+-------
+
+Any workload needs different volumes with different storage capacities
+to store its components. Following the total storage capacity needed,
+establish a number of volumes and their capacities for each node, as
+exemplified below:
+
+.. code-block:: yaml
+
+    metalk8s_lvm_lvs_vg_metalk8s: ['/dev/vdb']
```
Comment (NicolasT) [inline docs/usage/storage.rst]: This YAML seems ill-formatted.

Comment (lucieleonard) [reply docs/usage/storage.rst]: Why? I copy pasted the existing one a few versions ago

File: docs/usage/storage.rst [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+---------
+
+Volumes' capacities can be resized (one or several at once).
+Change the volume size value to a higher one and run:
+
+.. code::
+
+  ansible-playbook -i 'your inventory' -t storage playbooks/deploy.yml
+
+Both file system and logical volumes are automatically resized.
```
Comment (Zempashi) [inline docs/usage/storage.rst]: Tested yesterday, This doesn't work. I had to run 
```
resize2fs /dev/mapper/vg_metalk8s-lv01
```
To have my FS resized. But hold on with the doc, we'll try to have a PR that fix the behaviour and make it work like the docs states.

Comment (Zempashi) [reply docs/usage/storage.rst]: See #391

**Comment by NicolasT** [comment]
We'll want to squash most of this, I guess.

File: docs/usage/workload_storage.rst [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+.. code::
+
+  ansible-playbook -i <inventory>/hosts -t storage playbooks/deploy.yml
+
+Logical volumes are automatically resized.
+
+Configuration layout
+--------------------
+
+For every group or host created, the configuration files are merged.
```
Comment (Zempashi) [inline docs/usage/workload_storage.rst]: Group are not created that way (this is pure ansible documentation, not really covered yet). This is for applying configuration to groups and hosts.

File: docs/usage/workload_storage.rst [hunk 5]
```diff
@@ -0,0 +1,104 @@
+Workload Storage
+================
+
+Any workload needs different volumes with different storage capacities
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: Not all workloads require volumes: many services require no (local) state.

File: docs/usage/workload_storage.rst [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+to host its components. These volumes are stored in :term:`LVM LV`.
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: How can `volumes` `host` `components`?

Comment (NicolasT) [inline docs/usage/workload_storage.rst]: Consider not using `LVM LV` but `Logical Volume Manager (LVM) Logical Volumes (LVs)`. The reader may not be familiar with the terms at this point.

File: docs/usage/workload_storage.rst [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+Volumes
+-------
+
+Following the total storage capacity needed, establish a number of
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: How does this follow the 'total' storage capacity required? The 'number of volumes' and their respective sizes are based on the workload needs. Total storage capacity required depends on this number.

File: docs/usage/workload_storage.rst [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+            size: 11G
+        lv07:
+            size: 5G
+        lv08:
+            size: 5G
+
+Resize LVs
+----------
+
+Volumes' capacities can be resized (one or several at once).
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: Volumes can be resized. A volume's capacity is static, or rather simply the volume size. As in: what gets resized is the *volume*, not the *volume capacity*.

File: docs/usage/workload_storage.rst [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+Change the volume size value to a higher one and run:
+
+.. code::
+
+  ansible-playbook -i <inventory>/hosts -t storage playbooks/deploy.yml
+
+Logical volumes are automatically resized.
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: That's what was said above, no?

File: docs/usage/workload_storage.rst [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+Configuration layout
+--------------------
+
+The configuration can be applied to groups and hosts in two
+different ways.
+
+.. note::
+   Configuration files are merged for every created host or group.
+
+To apply a configuration, create a .yaml file in either (or both) group_vars
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]:
```
:file:`group_vars`
```

File: docs/usage/workload_storage.rst [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+and host_vars with the group name associated, or create a folder in group_vars
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]:
```
:file:`host_vars`
:file:`group_vars`
```

File: docs/usage/workload_storage.rst [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+or host_vars with several .yaml files.
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]:
```
:file:`host_vars`
```

File: docs/usage/workload_storage.rst [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+
+.. warning::
+   The ansible command above must be run at every file addition (deletions
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: It's `ansible-playbook`. Also,
```
:command:`ansible-playbook`
```

File: docs/usage/workload_storage.rst [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+   are not allowed).
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: What does this mean, 'deletions are not allowed'? Why couldn't you delete files from `group_vars` or `host_vars`? Of course you can.

I guess what's meant here is: LVs can't be removed. But that's a different context.

File: docs/usage/workload_storage.rst (hunk 13 above)
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: 'at every file addition': hmh? See below.

File: docs/usage/workload_storage.rst [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+
+Add extra LVs
+-------------
+
+It is possible to add extra lvm drive and volume for one node only.
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: `LVM`
`drives` and `volumes`, I guess.
`for one node` -> `on one node`, *or* `configure ... for one node only`.

File: docs/usage/workload_storage.rst [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+
+Exemplified below, a default storage configuration:
+
+`group_vars/kube-node/storage.yml`
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: Use `:file:` role.

File: docs/usage/workload_storage.rst [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+
+.. code::
+
+  metalk8s_lvm_vgs = ['vg_metalk8s']
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: This is the default. Why does it come up here and not in the file put in `group_vars` above?

File: docs/usage/workload_storage.rst [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+
+Volumes
+-------
+
+Following the total storage capacity needed, establish a number of
+volumes and their capacities for each node, as exemplified below:
+
+.. code-block:: yaml
+
+    metalk8s_lvm_lvs_vg_metalk8s:
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: Would still need to set PV drives.

File: docs/usage/workload_storage.rst [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+  metalk8s_lvm_drives_vg_metalk8s: ['/dev/vdb']
+  metalk8s_lvm_lvs_vg_metalk8s:
+    lv01:
+        size: 52G
+    lv02:
+        size: 52G
+    lv03:
+        size: 52G
+
+In host_vars, create a new file with:
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: `:file:`

File: docs/usage/workload_storage.rst [hunk 20]
```diff
@@ -0,0 +1,104 @@
...
+
+`host_vars/node_1.yml`
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: `:file:`

File: docs/usage/workload_storage.rst [hunk 21]
```diff
@@ -0,0 +1,104 @@
...
+   metalk8s_lvm_vgs = ['vg_metalk8s', 'mynewvg']
+   metalk8s_lvm_drives_mynewvg: ['/dev/vdc']
+   metalk8s_lvm_lvs_vg_metalk8s:
+     lv01:
+        size: 52G
+   metalk8s_lvm_lvs_mynewvg:
+     lv01:
+        size: 1T
+
+Except node_1, every machine has a single `vg_metalk8s` with six LVs
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: We should pick a role for node names etc.

File: docs/usage/workload_storage.rst [hunk 22]
```diff
@@ -0,0 +1,104 @@
...
+(three specified, three default).
+On node_1, there are two VGs (`vg_metalk8s` and `mynewvg`) with four LVs on
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]:
```
:term:`VG`
``` (or whatever the right term name is). Same for LVs.

**Review by NicolasT — CHANGES_REQUESTED** [review]

File: docs/usage/workload_storage.rst (hunk 7 above)
Comment (lucieleonard) [reply docs/usage/workload_storage.rst]: So I should just remove "total"? Or rework the whole sentence? I still don't quite get how capacities work

File: docs/usage/workload_storage.rst (hunk 21 above)
Comment (lucieleonard) [reply docs/usage/workload_storage.rst]: :service:?

File: docs/usage/workload_storage.rst (hunk 17 above)
Comment (lucieleonard) [reply docs/usage/workload_storage.rst]: Put a # in front of it. Is that right?

File: docs/usage/workload_storage.rst (hunk 7 above)
Comment (NicolasT) [reply docs/usage/workload_storage.rst]: I think the sentence needs to be reworked. What you aim at, I guess, is something like

> Based on all required storage volumes, create a configuration as below:

File: docs/usage/workload_storage.rst (hunk 5 above)
Comment (NicolasT) [reply docs/usage/workload_storage.rst]: How is this resolved?

File: docs/usage/workload_storage.rst (hunk 10 above)
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: It's `a YAML file`.

File: docs/usage/workload_storage.rst (hunk 12 above)
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: `YAML files`

File: docs/usage/workload_storage.rst (hunk 17 above)
Comment (NicolasT) [reply docs/usage/workload_storage.rst]: No, that would simply comment it out.

File: docs/usage/workload_storage.rst (hunk 21 above)
Comment (NicolasT) [reply docs/usage/workload_storage.rst]: Just put them between back-ticks for now, I guess.

File: docs/usage/workload_storage.rst [hunk 23]
```diff
@@ -0,0 +1,104 @@
...
+   metalk8s_lvm_vgs = ['vg_metalk8s', 'mynewvg']
+   metalk8s_lvm_drives_mynewvg: ['/dev/vdc']
+   metalk8s_lvm_lvs_vg_metalk8s:
+     lv01:
+        size: 52G
+   metalk8s_lvm_lvs_mynewvg:
+     lv01:
+        size: 1T
+
+Except node_1, every machine has a single `vg_metalk8s` with six logical
```
Comment (lucieleonard) [inline docs/usage/workload_storage.rst]: :service:`node`?

File: docs/usage/workload_storage.rst [hunk 24]
```diff
@@ -0,0 +1,104 @@
+Workload Storage
+================
+
+Some workloads need different volumes with different storage capacities
+to fit its components needs. These volumes are stored in Logical Volumes
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: I believe you could use
```
:term:`LVM Logical Volumes <LVM LV>`
```

But would need to test.

File: docs/usage/workload_storage.rst [hunk 25]
```diff
@@ -0,0 +1,104 @@
...
+The configuration can be applied to groups and hosts in two
+different ways.
+
+.. note::
+   Configuration files are merged for every created host or group.
+
+To apply a configuration, create a .yaml file in either (or both)
+:file:`group_vars` and :file:`host_vars` with the group name
+associated, or create a folder in :file:`group_vars` or
+:file:`host_vars` with several .yaml files.
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: I don't like the use of `.yaml files`, would prefer `YAML files` (various occurrences).

File: docs/usage/workload_storage.rst [hunk 26]
```diff
@@ -0,0 +1,104 @@
...
+The :command:`Ansible-playbook` above must be run.
```
Comment (NicolasT) [inline docs/usage/workload_storage.rst]: `ansible-playbook`, no capital.

File: docs/usage/workload_storage.rst (hunk 24 above)
Comment (lucieleonard) [reply docs/usage/workload_storage.rst]: It works :)

File: docs/usage/workload_storage.rst (hunk 23 above)
Comment (NicolasT) [reply docs/usage/workload_storage.rst]: Just `node_1` for now I guess.

**Comment by NicolasT** [comment]
OK. Let's get together to
- Squash all commits
- Rebase on `development/1.0`
- Retarget this PR to `development/1.0`
and it can go in.

**Comment by NicolasT** [comment]
@bert-e reset

**Review by NicolasT — APPROVED** [review]
LGTM :+1:

**Comment by lucieleonard** [comment]
@bert-e status

**Comment by NicolasT** [comment]
@bert-e status

**Pull request merged**